### ISSUE TITLE
[docs] Improve the error message of `naming-convention/use-state`

### ISSUE THREAD

**Issue by bhollis** [Rel1cx/eslint-react#980]
### Describe the bug

```ts
const [{ result, processing }, setState] = useState<ProcessState>({
  processing: false,
  resultStoreId: selectedStore.id,
  result: null,
});
```

This triggers "An useState call is not destructured into value + setter pair  @eslint-react/naming-convention/use-state" but it shouldn't.

### Reproduction

```ts
const [{ result, processing }, setState] = useState<ProcessState>({
  processing: false,
  resultStoreId: selectedStore.id,
  result: null,
});
```

### Expected behavior

No error.

### Platform and versions

```Text
Node v23.5.0
ESLint 9.19.0
@eslint-react/react-plugin 1.31.0
```

### Stack trace

```Text

```

**Comment by bhollis** [comment]
I also get a warning for this code:

```ts
const [liveQueryLive, setLiveQuery] = useState(searchQuery ?? '');
```

**Comment by bhollis** [comment]
And this one:

```ts
const [options, setOptionsState] = useState<ItemPickerState>();
```

**Comment by Rel1cx** [comment]
The rule in the issue title should be `use-state`, not `no-use-context`.

**Comment by Rel1cx** [comment]
The code in the example above doesn't have a false positive, but the error message here has room for improvement. The current message might mislead users into thinking they're not using destructuring assignment.
